ref(ds): Specify how the DSC is interpreted in Relay (#17899)
getsentry/sentry-docs: develop-docs/application-architecture/dynamic-sampling/architecture.mdx

1. Rule `1` is matched against the DSC, since it is of type `trace`. The `samplingValue` is a `factor` with value `2.0`.
2. Because rule `1` was a factor rule, the matching continues and rule `2` will again be matched against the DSC, since it is of type `trace`. The `samplingValue` is a `sampleRate`, thus the matching will stop and the sample rate will be computed as `2.0 * 0.5 = 1.0`, where `2.0` is the factor accumulated from the previous rule and `0.5` is the sample rate of the current rule.

### Interpreting the Dynamic Sampling Context

The existence of a dynamic sampling context does not necessarily mean it is valid, Relay differentiates between three cases:
1. No dynamic sampling context
2. A dynamic sampling context originating from the same organization

> 3. A dynamic sampling context originating in a project of a different organization or from an unknown project.

Reviewer: ```suggestion
The existence of a dynamic sampling context does not necessarily mean it is valid. Relay differentiates between three cases:
1. No dynamic sampling context
2. A dynamic sampling context originating in a project of the same organization
3. A dynamic sampling context originating in a project of a different organization or an unknown project.
```

> If an envelope received by Relay does not contain a dynamic sampling context it is always sampled. Certain payloads do always require a dynamic sampling context to be present and will be rejected without a dynamic sampling context.

Reviewer: I realize that the second sentence is meant to describe an exception to the first, but they sound contradictory. Perhaps it would help to be explicit about which payloads require a DSC.

Author: I decided against having an exhaustive list here, these lists go out of date quickly and I'd prefer this document to describe methodology not implementation.

Reviewer: In that case, I would suggest something like
```suggestion
Certain payloads do always require a dynamic sampling context to be present and will be rejected without a dynamic sampling context. All other payloads will always be sampled if they do not contain a dynamic sampling context.
```

Author: Reworded things a bit.

Reviewer: Much better, thank you!

A dynamic sampling context which originates from either the same project or a project within the same organization is considered valid and Relay will apply the sampling rules from the root project as described in the previous section.

> DSCs originating in different organizations or unknown projects are discarded and Relay will re-compute a DSC based on the data of the payload and scoped to the current project. The computed dynamic sampling context is then used to apply the dynamic sampling rules.

Reviewer 2: I think this would be easier to parse if we inline the descriptions into the list, and make the headers bold. 
```suggestion
1. **No dynamic sampling context**: If an envelope received by Relay does not contain a dynamic sampling context it is always sampled, unless the payload requires a DSC to always be present.

2. **A dynamic sampling context originating in a project of the same organization**: A dynamic sampling context which originates from either the same project or a project within the same organization is considered valid and Relay will apply the sampling rules from the root project as described in the previous section.

3. **A dynamic sampling context originating in a project of a different organization or an unknown project**: DSCs originating in different organizations or unknown projects are discarded and Relay will re-compute a DSC based on the data of the payload and scoped to the current project. The computed dynamic sampling context is then used to apply the dynamic sampling rules.

```

> DSCs originating in different organizations or unknown projects are discarded and Relay will re-compute a DSC based on the data of the payload and scoped to the current project. The computed dynamic sampling context is then used to apply the dynamic sampling rules.

Reviewer 3: Is this the only case where we generate a DSC from the payload? Or do we also do it if the DSC is missing but not required?

Author: Should be the only case, no DSC ends up at 100% SR
